3rd-party/class.jetpack-amp-support.php

```diff
@@ -41,8 +41,14 @@ public static function init() {
 		// Post rendering changes for legacy AMP.
 		add_action( 'pre_amp_render_post', array( 'Jetpack_AMP_Support', 'amp_disable_the_content_filters' ) );
 
+		// Disable Comment Likes.
+		add_filter( 'jetpack_comment_likes_enabled', array( 'Jetpack_AMP_Support', 'comment_likes_enabled' ) );
+
 		// Transitional mode AMP should not have comment likes.
-		add_action( 'the_content', array( 'Jetpack_AMP_Support', 'disable_comment_likes_before_the_content' ) );
+		add_filter( 'the_content', array( 'Jetpack_AMP_Support', 'disable_comment_likes_before_the_content' ) );
+
+		// Remove the Likes button from the admin bar.
+		add_filter( 'jetpack_admin_bar_likes_enabled', array( 'Jetpack_AMP_Support', 'disable_likes_admin_bar' ) );
```

Comment:
Can these filters not be deferred to happen at the `wp` action? Or is that too late?

Reply:
Not needed if the filters will only ever apply after the `wp` action or if they run in an admin context. Otherwise, if there is a chance they can be applied on the frontend before the `wp` then the `is_amp_endpoint()` function will complain about being called too early.

Reply:
We should be safe there. 👍